### PR TITLE
[features] Refactoring and proper Eigen usage (5/N)

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

File: features/include/pcl/features/impl/fpfh.hpp [hunk 1]
```diff
@@ -161,12 +163,14 @@ pcl::FPFHEstimation<PointInT, PointNT, PointOutT>::weightPointSPFHSignature (
...
-  for (int f1_i = 0; f1_i < nr_bins_f1; ++f1_i)
-    fpfh_histogram[f1_i] *= static_cast<float> (sum_f1);
-  for (int f2_i = 0; f2_i < nr_bins_f2; ++f2_i)
-    fpfh_histogram[f2_i + nr_bins_f1] *= static_cast<float> (sum_f2);
-  for (int f3_i = 0; f3_i < nr_bins_f3; ++f3_i)
-    fpfh_histogram[f3_i + nr_bins_f12] *= static_cast<float> (sum_f3);
+  const auto denormalize_with = [](const auto& factor)
+    {
+      return [&factor](const auto& data) { return data * static_cast<float> (factor); };
+    };
```
Comment (taketwo) [inline features/include/pcl/features/impl/fpfh.hpp]: What about:

```cpp
const auto denormalize_with = [](float factor) {
  return [=](const auto& data) { return data * factor; };
};
```

File: features/include/pcl/features/impl/grsd.hpp [hunk 2]
```diff
@@ -92,21 +92,21 @@ pcl::GRSDEstimation<PointInT, PointNT, PointOutT>::computeFeature (PointCloudOut
   // Save the type of each point
   int NR_CLASS = 5; // TODO make this nicer
   std::vector<int> types (radii->points.size ());
-  for (std::size_t idx = 0; idx < radii->points.size (); ++idx)
-    types[idx] = getSimpleType (radii->points[idx].r_min, radii->points[idx].r_max);
+  std::transform(radii->points.begin (), radii->points.end (), types.begin (),
```
Comment (taketwo) [inline features/include/pcl/features/impl/grsd.hpp]: Can we use const here?

```suggestion
  std::transform(radii->points.cbegin (), radii->points.cend (), types.begin (),
```

File: features/include/pcl/features/impl/fpfh.hpp [hunk 3]
```diff
@@ -161,12 +163,14 @@ pcl::FPFHEstimation<PointInT, PointNT, PointOutT>::weightPointSPFHSignature (
     sum_f3 = 100.0 / sum_f3;           // histogram values sum up to 100
 
   // Adjust final FPFH values
-  for (int f1_i = 0; f1_i < nr_bins_f1; ++f1_i)
-    fpfh_histogram[f1_i] *= static_cast<float> (sum_f1);
-  for (int f2_i = 0; f2_i < nr_bins_f2; ++f2_i)
-    fpfh_histogram[f2_i + nr_bins_f1] *= static_cast<float> (sum_f2);
-  for (int f3_i = 0; f3_i < nr_bins_f3; ++f3_i)
-    fpfh_histogram[f3_i + nr_bins_f12] *= static_cast<float> (sum_f3);
+  const auto denormalize_with = [](const auto& factor)
```
Comment (kunaltyagi) [inline features/include/pcl/features/impl/fpfh.hpp]:
```suggestion
const auto denormalize_with = [](float factor) {
  return [=](const auto& data) { return data * factor; };
};
```

File: features/include/pcl/features/impl/fpfh.hpp (hunk 1 above)
Comment (kunaltyagi) [reply features/include/pcl/features/impl/fpfh.hpp]: Looks nicer.

**Pull request merged**